### PR TITLE
Added tests for NetworkController

### PR DESCRIPTION
**Pull request by bkonyi**

### CODE REVIEW

File: packages/devtools_app/test/network_controller_test.dart [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+
+      test(
+        'start and pause recording',
+        () async {
+          expect(
+            controller.isPolling,
+            false,
+          );
+          final notifier = controller.recordingNotifier;
+          await addListenerScope(
```
Comment (kenzieschmoll) [inline packages/devtools_app/test/network_controller_test.dart]: Couldn't the test still pass if the listener was never called?

File: packages/devtools_app/lib/src/network/network_controller.dart [hunk 2]
```diff
@@ -16,13 +16,15 @@ class NetworkController {
   }
 
   /// Notifies that new HTTP requests have been processed.
-  ValueListenable get requestsNotifier => _httpRequestsNotifier;
+  ValueListenable<HttpRequests> get requestsNotifier => _httpRequestsNotifier;
```
[the remark this answers is not in the document]
Comment (bkonyi) [reply packages/devtools_app/lib/src/network/network_controller.dart]: If you access `notifier.value` without specifying a type it's inferred as `dynamic`, messing with IDE auto-completion.

File: packages/devtools_app/test/network_controller_test.dart (hunk 1 above)
Comment (bkonyi) [reply packages/devtools_app/test/network_controller_test.dart]: Yes, good point. I've added a check in `addListenerScope` to ensure the listener is actually called.

File: packages/devtools_app/test/network_controller_test.dart [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+          controller = NetworkController();
+        },
+      );
+
+      test(
+        'initialize recording state',
+        () async {
+          expect(
+            controller.isPolling,
+            false,
```
[the remark this answers is not in the document]
Comment (bkonyi) [reply packages/devtools_app/test/network_controller_test.dart]: That makes sense. I think I've gotten them all in my latest commit.

File: packages/devtools_app/test/network_controller_test.dart [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+    'NetworkController',
+    () {
+      NetworkController controller;
+      FakeServiceManager fakeServiceManager;
+      Timeline timeline;
+
+      setUpAll(
+        () async {
+          const testDataPath =
+              '../devtools_testing/lib/support/http_request_timeline_test_data.json';
```
[the remark this answers is not in the document]
Comment (bkonyi) [reply packages/devtools_app/test/network_controller_test.dart]: Added a TODO to do this later.

**Pull request merged**